### PR TITLE
CRIU restore setup Xtrace output file

### PR DESCRIPTION
**Pull request by JasonFengJ9**
Add support for `-Xtrace:maximal=mt,methods=java/lang/System.setProperties(),output=traceOutput.txt`;
Added a test.

Signed-off-by: Jason Feng <fengj@ca.ibm.com>

### CODE REVIEW

File: runtime/rastrace/trcengine.c [hunk 1]
```diff
@@ -1011,13 +1012,16 @@ criuRestoreInitializeTrace(J9VMThread *thr)
...
 
-	if (J9VMDLLMAIN_OK == traceInitializationHelper(vm, UT_THREAD_FROM_VM_THREAD(thr), vm->checkpointState.restoreArgsList, TRUE)) {
+	if (J9VMDLLMAIN_OK == traceInitializationHelper(vm, tempThr, vm->checkpointState.restoreArgsList, TRUE)) {
 		RasGlobalStorage *j9ras = (RasGlobalStorage *)vm->j9rasGlobalStorage;
 		if ((NULL != j9ras->traceMethodTable)
 			|| (NULL != j9ras->triggerOnMethods)
 		) {
-			if (OMR_ERROR_NONE == enableMethodTraceHooks(vm)) {
+			if (OMR_ERROR_NONE == enableMethodTraceHooks(vm)
+				&& (OMR_ERROR_NONE == setupTraceWorkerThread(tempThr))
```
Comment (tajila) [inline runtime/rastrace/trcengine.c:1023]: do you do a shutdown of the worker thread pre-checkpoint?

Comment (JasonFengJ9) [reply runtime/rastrace/trcengine.c:1023]: The `setupTraceWorkerThread()` does `openTraceFile()` for `externalTrace` & `extExceptTrace`. 
The matching shutdown function is `cleanupTraceWorkerThread()` which is only invoked when there were `externalTrace/extExceptTrace` files opened, which is not expected to be a typical usage of pre-checkpoint.

There is no straightforward way to invoke `cleanupTraceWorkerThread()` before a checkpoint. I would open another PR to address that scenario which will also fix the restore issue if an `Xtrace` output file before the checkpoint is modified or deleted though this is an unlikely case in a container environment.
I will add a doc item about this restriction.

**Review by tajila — APPROVED** [review]

**Comment by tajila** [comment]
@JasonFengJ9 Please rebase these changes

**Comment by JasonFengJ9** [comment]
@tajila just rebased.

**Comment by tajila** [comment]
jenkins test sanity xlinux jdk17

**Pull request merged**